de/nmichael/efa/data/efacloud/TableBuilder.java

```diff
@@ -215,6 +212,25 @@ public class TableBuilder {
         allowedMismatches.put("efa2statistics", 0);  // All may change
     }
     public static final String fixid_allowed = "efa2logbook efa2messages efa2boatdamages efa2boatreservations";
+    public static final ArrayList<String> tablenames = new ArrayList<>();
```

Comment:
Arraylist has a very poor performance when it comes to searching.
This should be converted into an HashTable <String>.